nixos/modules/security/setuid-swap.c

```diff
@@ -0,0 +1,36 @@
+#define _GNU_SOURCE
```

Comment:
Hmm, normally I'd agree, but IMO this is widespread enough not to need it. From https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html#index-_005fGNU_005fSOURCE:
> If you define this macro, everything is included: ISO C89, ISO C99, POSIX.1, POSIX.2, BSD, SVID, X/Open, LFS, and GNU extensions. In the cases where POSIX.1 conflicts with BSD, the POSIX definitions take precedence.

In other words, "this is non-portable, give me all the extensions"